firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -207,17 +207,53 @@ public void finalizeReports(String currentSessionId) {
             TRANSFORM.reportToJson(report.withEvents(ImmutableList.from(events))));
       }
       recursiveDelete(sessionDirectory);
+      capFinalizedReports();
     }
   }
 
-  public List<CrashlyticsReport> loadFinalizedReports() {
-    final List<CrashlyticsReport> allReports = new ArrayList<>();
-    final List<File> priorityReports = getAllFilesInDirectory(priorityReportsDirectory);
-    for (File reportFile : priorityReports) {
-      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+  private void capFinalizedReports() {
+    List<File> allReportFiles = getAllFinalizedReportFiles();
+    if (allReportFiles.size() > defaultMaxReportsToKeep) {
```

Comment:
Any chance we could extract a variable for `size` here, outside the `if` to clarify that the same value should be used in the check and in the `subList` call?

Reply:
sure